src/core/core/rpc_engine.cpp

```diff
@@ -846,14 +850,28 @@ namespace dsn {
                             {
                                 resolver->on_access_failure(req2->header->gpid.u.partition_index, err);
 
+                                /*
+                                // TODO(qinzuoyan): because the user will fetch reply states from this response_task 'call',
```

Comment:
I think this is way too heavy, and we don't want to expose those to core-outsides. The good news is that this piece of code is inside the core, so we should be able to have a design transparent to the outside.